src/DonorDashboards/App.php

```diff
@@ -180,4 +181,36 @@ public function loadAssets() {
 			GIVE_VERSION
 		);
 	}
+
+	/**
+	 * Determine if the login should be enabled.
+	 *
+	 * @unreleased
+	 *
+	 * @return bool
+	 */
+	protected function loginEnabled() {
+		// We need to get all the form IDs.
+		$formIds = get_posts(
+			[
+				'fields' => 'ids',
+				'numberposts' => -1,
+				'post_status' => 'publish',
+				'post_type' => 'give_forms',
+			]
+		);
+
+		// By default, the login is disabled.
+		$loginEnabled = false;
+		foreach ( $formIds as $formId ) {
+			if ( give_show_login_register_option( $formId ) !== 'none' ) {
+				// Once there is a single form that it is enabled, we can bail out
+				// early since the login needs to be enabled.
+				$loginEnabled = true;
+				break;
+			}
```

Comment:
```suggestion
				if (
				give_logged_in_only( $formId ) &&
				give_show_login_register_option( $formId ) === 'none'
			) {
				// Once there is a single form that it is enabled, we can bail out
				// early since the login needs to be enabled.
				$loginEnabled = true;
				break;
			}
```
I suggest the above code find out correct (whether need logic or not ) donor access to donation form

Reply:
@ravinderk Can you explain why this is necessary?

Reply:
@knowler ignore this.